### PR TITLE
Timur katimullin 41 50 tasks

### PR DESCRIPTION
**Pull request by linkouth**
Adds 41-50 tasks solutions.

### CODE REVIEW

File: tasks/41_sort-by-bits/sort-by-bits.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/**
+ * sortByBits - sort an array of 32-bit integers
```
Comment (vovcyan) [inline tasks/41_sort-by-bits/sort-by-bits.js:2]: Please add example

**Review by vovcyan — CHANGES_REQUESTED** [review]

File: tasks/44_time-simulation/time-simulation.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/**
+ * class SimTime
+ */
+export class SimTime {
```
Comment (vovcyan) [inline tasks/44_time-simulation/time-simulation.js]: See description of task 44

function SimTime() {
  // ...
}

let sim = new SimTime();

**Review by vovcyan — CHANGES_REQUESTED** [review]

File: tasks/44_time-simulation/time-simulation.js (hunk 2 above)
Comment (linkouth) [reply tasks/44_time-simulation/time-simulation.js]: Fixed.

File: tasks/41_sort-by-bits/sort-by-bits.js (hunk 1 above)
Comment (linkouth) [reply tasks/41_sort-by-bits/sort-by-bits.js:2]: Added examples for every functions.

File: tasks/44_time-simulation/time-simulation.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  this.currentRealtime = 0;
+
+  /**
+   * get - returns the current time withing the simulation
+   * @example
+   * sim.get();
+   * //return 10
+   * @return {Number} time withing the simulation
+   */
+  this.get = () => {
```
Comment (alexkond) [inline tasks/44_time-simulation/time-simulation.js]: Please read about using `prototype` for adding methods to the class and why it is a better way to do this.

File: tasks/45_equal-sides/equal-sides.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ * the integers to the left of N is equal to the sum
+ * of the integers to the right of N
+ * @example
+ * findEqualIndex([1, 2, 3, 4, 3, 2, 1])
+ * // return 3
+ * @param {Array<Number>} arr
+ * @return {Number}
+ */
+export function findEqualIndex(arr) {
+  if (arr.length < 2) return 0;
```
Comment (alexkond) [inline tasks/45_equal-sides/equal-sides.js]: In JS it's a best practice to use curly brackets even for single-line blocks of code.

File: tasks/47_string-expansion/string-expansion.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ * @example
+ * stringExpansion('3D2a5d2f');
+ * // return 'DDDaadddddff'
+ * @param {String} str
+ * @return {String}
+ */
+export function stringExpansion(str) {
+  let result = '';
+  let repeatCount = 0;
+  const isLetter = (arg) => {
```
Comment (alexkond) [inline tasks/47_string-expansion/string-expansion.js]: This function could be declared in the outer scope.

**Review by alexkond — CHANGES_REQUESTED** [review]

File: tasks/44_time-simulation/time-simulation.js (hunk 3 above)
Comment (linkouth) [reply tasks/44_time-simulation/time-simulation.js]: Fixed.

File: tasks/45_equal-sides/equal-sides.js (hunk 4 above)
Comment (linkouth) [reply tasks/45_equal-sides/equal-sides.js]: Refactored.

File: tasks/47_string-expansion/string-expansion.js (hunk 5 above)
Comment (linkouth) [reply tasks/47_string-expansion/string-expansion.js]: Fixed.

File: tasks/47_string-expansion/string-expansion.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+ * stringExpansion('3D2a5d2f');
+ * // return 'DDDaadddddff'
+ * @param {String} str
+ * @return {String}
+ */
+export function stringExpansion(str) {
+  let result = '';
+  let repeatCount = 0;
+  for (let i = 0; i < str.length; i++) {
+    if (isNotNumber(str[i])) {
```
Comment (alexkond) [inline tasks/47_string-expansion/string-expansion.js]: Looks like `isNotNumber()` function just parse string to number and checks if it is `NaN`. But later in the code, you parse `str[i]` again. Why not parse it in the beginning of the loop and save in variable, use `isNaN` here and later access it via variable, not parsing again?

**Review by alexkond — CHANGES_REQUESTED** [review]

**Review by alexkond — APPROVED** [review]

**Review by vovcyan — APPROVED** [review]

**Pull request merged**